openmdao/visualization/meta_model_viewer/meta_model_visualization.py

```diff
@@ -0,0 +1,554 @@
+"""Define output of Meta Models and visualize the results."""
+
+from collections import OrderedDict
+import math
+from bokeh.io import curdoc
+from bokeh.layouts import row, column
+from bokeh.plotting import figure
+from bokeh.models import Slider, ColumnDataSource
+from bokeh.models import ColorBar, BasicTicker, LinearColorMapper, Range1d
+from bokeh.models.widgets import TextInput, Select
+
+# Misc Imports
+from scipy.spatial import cKDTree
+import numpy as np
+
+
+def stack_outputs(outputs_dict):
+    """
+    Stack the values of a dictionary.
+
+    Parameters
+    ----------
+    outputs_dict : dict
+        Dictionary of outputs
+
+    Returns
+    -------
+    array
+        np.stack of values
+    """
+    output_lists_to_stack = []
+    for values in outputs_dict.values():
+        output_lists_to_stack.append(np.asarray(values))
+
+    return np.stack(output_lists_to_stack, axis=-1)
+
+
+class MetaModelVisualization(object):
+    """
+    Top-level container for the Meta Model Visualization.
+
+    Attributes
+    ----------
+    prob : om.Problem
+        Name of variable corresponding to Problem Component
+    surrogate_comp : MetaModel
+        Name of variable corresponding to Meta Model Component
+    resolution : int
```

Comment:
both `prob` and `surrogate_comp` attributes are just object references rather than "name of variable corresponding to..."
also Problem is not a Component...